app/models/discussion.rb

```diff
@@ -0,0 +1,40 @@
+# Model class for discussions (used to group comments)
+class Discussion < ApplicationRecord
+  belongs_to :resource, inverse_of: :discussion, polymorphic: true
+
+  has_many :comments,
+           class_name: 'Comment',
+           foreign_key: :discussion_id,
+           inverse_of: :discussion,
+           dependent: :destroy
+
+  # Instance methods
+
+  def discussion_hidden
+    hidden
+  end
+
+  def discussion_locked
+    locked
+  end
```

Comment:
See comment above, `discussion_hidden` and `discussion_locked` can be deleted if `prefix` option used with `delegate`.